eng/scripts/packagevalidation/PrePublish-Validation.ps1

```diff
@@ -0,0 +1,97 @@
+[CmdletBinding()]
+Param (
+  [ValidateNotNullOrEmpty()]
+  [string] $PackagePath,
+  [string] $ArtifactName,
+  [string] $WorkingDirectory
+)
+
+. "${PSScriptRoot}\..\common\scripts\logging.ps1"
+. "${PSScriptRoot}\..\common\scripts\SemVer.ps1"
+
+$WinVersion = (Get-ChildItem -Path "C:\Program Files (x86)\Windows Kits\10\bin\10.0*" | Sort-Object -Descending)[0]
+$SignTool = "C:\Program Files (x86)\Windows Kits\10\bin\$($WinVersion.Name)\x64\signtool.exe"
+
+New-Item -Path $WorkingDirectory -Name "Validation" -ItemType "directory"
+$ValidationDirectory = "$WorkingDirectory\Validation"
+
+Copy-Item -Path $PackagePath -Destination "$ValidationDirectory\package.zip"
+Expand-Archive -LiteralPath "$ValidationDirectory\package.zip" "$ValidationDirectory\extracted"
+
+$PackageName = Split-Path -Path $PackagePath -Leaf
+$PackageDlls = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.dll" -Recurse
+$NuspecFile = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.nuspec" -Recurse
+
+$NuspecContent = new-object xml
+$NuspecContent.Load($NuspecFile[0].FullName)
+$PackageId = $NuspecContent.package.metadata.id
+$PackageVersion = $NuspecContent.package.metadata.version
+$ProjectUrl = $NuspecContent.package.metadata.projectUrl
+$Description = $NuspecContent.package.metadata.description
+
+$IsValidPackage = $true
+
+LogDebug "Validating that Binaries and Packages are Signed..."
+foreach ($file in $PackageDlls) {
+    &$SignTool verify /pa $file.FullName
+    if ($LASTEXITCODE -ne 0) {
+        LogError "Validation Failed. Library [$($file.FullName)] is not signed."
+        $IsValidPackage = $false
+    }
+}
+
+LogDebug "Validating that Package Name matches Guidelines..."
+try {
+    $Version = ([AzureEngSemanticVersion]::ParseVersionString($PackageVersion)).ToString()
```

Comment:
while this might work for our track 2 packages. Packages going through our partner release pipelines might not follow this exact versioning scheme.

Reply:
Is there a better way I can validate the packageVersion?

Reply:
I don't know that there is really a better way we can simply stick with this for now and see if we hit issues.